Lib/test/datetimetester.py

```diff
@@ -399,6 +400,11 @@ def test_harmless_mixed_comparison(self):
         self.assertIn(me, [1, 20, [], me])
         self.assertIn([], [me, 1, 20, []])
 
+        # Comparison over a type other that is not object's type should return
+        # NotImplemented and fallback to other.__eq__. In this case ANY.__eq__
+        # always returns True.
+        self.assertEqual(me, ANY)
```

Comment:
Is this a guaranteed property of `mock.ANY`? It doesn't seem to be [documented here](https://docs.python.org/3/library/unittest.mock.html#any).

In `dateutil` I have a [dedicated test object for this](https://github.com/dateutil/dateutil/blob/master/dateutil/test/_common.py#L236). If "is an object that compares equal to everything" is not a documented property of `mock.ANY`, it might be worth just creating a test object specifically for this purpose.

Reply:
`mock.ANY` is to be used to in the context of mock helper comparisons where always returns True for all objects. So it is like a copy of the test object used in dateutil. I guess I will just create a similar one for the test and don't want to use undocumented usage of being it outside mock helpers contradicting myself https://bugs.python.org/issue37555#msg347731 :)

https://github.com/python/cpython/blob/994a3b88dca852696351358e2743313e546b5ecf/Lib/unittest/mock.py#L2263-L2270